ugs-pendant/src/main/java/com/willwinder/universalgcodesender/pendantui/PendantUI.java

```diff
@@ -55,65 +58,82 @@
+
+        ContextHandler pendantResourceHandlerContext = new ContextHandler();
+        pendantResourceHandlerContext.setContextPath("/");
+        pendantResourceHandlerContext.setHandler(pendantResourceHandler);
+
+
+        ResourceHandler oldPendantResourceHandler = new ResourceHandler();
+        oldPendantResourceHandler.setDirectoriesListed(true);
+        oldPendantResourceHandler.setWelcomeFiles(new String[]{"index.html"});
+        oldPendantResourceHandler.setBaseResource(getBaseResource("/pendantUI/old"));
+
+        ContextHandler oldPendantResourceHandlerContext = new ContextHandler();
+        oldPendantResourceHandlerContext.setContextPath("/old");
+        oldPendantResourceHandlerContext.setHandler(oldPendantResourceHandler);
+
 
         ContextHandler sendGcodeContext = new ContextHandler();
         sendGcodeContext.setContextPath("/sendGcode");
-        sendGcodeContext.setBaseResource(getBaseResource());
         sendGcodeContext.setClassLoader(Thread.currentThread().getContextClassLoader());
         sendGcodeContext.setHandler(new SendGcodeHandler());
 
         ContextHandler adjustManualLocationContext = new ContextHandler();
         adjustManualLocationContext.setContextPath("/adjustManualLocation");
-        adjustManualLocationContext.setBaseResource(getBaseResource());
         adjustManualLocationContext.setClassLoader(Thread.currentThread().getContextClassLoader());
         adjustManualLocationContext.setHandler(new AdjustManualLocationHandler());
 
         ContextHandler getSystemStateContext = new ContextHandler();
         getSystemStateContext.setContextPath("/getSystemState");
-        getSystemStateContext.setBaseResource(getBaseResource());
         getSystemStateContext.setClassLoader(Thread.currentThread().getContextClassLoader());
         getSystemStateContext.setHandler(new GetSystemStateHandler());
 
         ContextHandler configContext = new ContextHandler();
         configContext.setContextPath("/config");
-        configContext.setBaseResource(getBaseResource());
         configContext.setClassLoader(Thread.currentThread().getContextClassLoader());
         configContext.setHandler(new ConfigHandler());
         configContext.setInitParameter("cacheControl", "max-age=0, public");
 
-        HandlerList handlers = new HandlerList();
-        handlers.setHandlers(new Handler[] {configContext, sendGcodeContext, adjustManualLocationContext, getSystemStateContext, resourceHandler, new DefaultHandler()});
+        // Create a servlet servletContextHandler
+        ServletContextHandler servletContextHandler = new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
+        servletContextHandler.setContextPath("/api");
+        ServletHolder servletHolder = servletContextHandler.addServlet(ServletContainer.class, "/*");
+        servletHolder.setInitOrder(1);
+        servletHolder.setInitParameter("javax.ws.rs.Application", AppConfig.class.getCanonicalName());
```

Comment:
I see, so this is why it's difficult to inject the backend object.

Reply:
It was some time ago that I tried this out. But as I remember I couldn't use the constructor fetch the constructor. This was the cleanest way to get it working.